xbmc/filesystem/DirectoryFactory.cpp

```diff
@@ -92,7 +92,7 @@ using namespace XFILE;
  \return IDirectory object to access the directories on the share.
  \sa IDirectory
  */
-IDirectory* CDirectoryFactory::Create(const CURL& url)
+IDirectory* CDirectoryFactory::Create(const CURL& url, const std::any& hint)
```

Comment:
i find hard understand what is this `std::any hint`,
unless an alternative to using `std::any` could be found,
at least should be specified on the documentation for what type of protocol this parameter is used and what type of data is accepted
(if so the documentation above should be also moved to the header)

Reply:
I figured out a way to do it using the existing Directory api. Please take a look at the latest push.